src/Fields/fieldvector.jl

```diff
@@ -179,6 +178,41 @@ end
     return dest
 end
 
+# Recursively call transform_bc_args() on broadcast arguments in a way that is statically reducible by the optimizer
+# see Base.Broadcast.preprocess_args
+@inline transform_bc_args(args::Tuple, inds...) = (
+    transform_broadcasted(args[1], inds...),
+    transform_bc_args(Base.tail(args), inds...)...,
+)
+@inline transform_bc_args(args::Tuple{Any}, inds...) =
+    (transform_broadcasted(args[1], inds...),)
+@inline transform_bc_args(args::Tuple{}, inds...) = ()
+
+function transform_broadcasted(
+    bc::Base.Broadcast.Broadcasted{FieldVectorStyle},
+    symb,
+    axes,
+)
+    Base.Broadcast.Broadcasted(
+        bc.f,
+        transform_bc_args(bc.args, symb, axes),
+        axes,
+    )
+end
+transform_broadcasted(fv::FieldVector, symb, axes) =
+    parent(getproperty(fv, symb))
+transform_broadcasted(x, symb, axes) = x
+@inline function Base.copyto!(
+    dest::FieldVector,
+    bc::Base.Broadcast.Broadcasted{FieldVectorStyle},
+)
+    for symb in propertynames(dest)
+        p = parent(getproperty(dest, symb))
```

Comment:
I think here you can call the internal field vector function `backing_array`

Reply:
Yeah, I just changed this to `p = parent(getfield(_values(dest), symb))` and that seems to work.

Reply:
I think calling `backing_array` would still be problematic, because even though you avoid calling `parent`, the `getproperty` call `unwrap`s the data and we get a method error with `copyto!(::Float64, ...)`